src/hooks/usePagination.js

```diff
@@ -37,6 +38,7 @@ export const usePagination = props => {
     ]
   } = props
 
+  const rowDep = disablePageResetOnDataChange ? null : rows
```

Comment:
note that this construct is to ensure that lint doesn't give this error:

```
warning  React Hook useLayoutEffect has a second argument which is not an array literal. This means we can't statically verify whether you've passed the correct dependencies
```
when I tried to inline the dep change.